prow/cmd/hook/events.go

```diff
@@ -20,40 +20,9 @@ import (
 	"github.com/Sirupsen/logrus"
 
 	"k8s.io/test-infra/prow/github"
-	"k8s.io/test-infra/prow/plugins"
 )
 
-// EventAgent pulls events off of the queues and dispatches to the relevant
-// plugins.
-type EventAgent struct {
-	Plugins *plugins.PluginAgent
-
-	PullRequestEvents  <-chan github.PullRequestEvent
-	IssueCommentEvents <-chan github.IssueCommentEvent
-	StatusEvents       <-chan github.StatusEvent
-}
-
-// Start starts listening for events. It does not block.
-func (ea *EventAgent) Start() {
-	go func() {
-		for pr := range ea.PullRequestEvents {
-			go ea.handlePullRequestEvent(pr)
```

Comment:
Yeah, that should be good. 
The existing plugins don't seem to have any races, so a warning for the new plugins should be fine. If the github client had races, we'd have seen it already.